### ISSUE TITLE
Hardcoded ReturnUrl in the Delegation Mode

### ISSUE THREAD

**Issue by RetoRechsteiner** [Azure/api-management-developer-portal#465]
## Bug description

When I do a delegated login form an URL like https://**service-name**.developer.azure-api.net/products the ReturnUrl from the delegation is hardcoded to "/".

## Reproduction steps

See row 26 in the userSigninViewModelBinder.ts.
-> https://github.com/Azure/api-management-developer-portal/blob/34f3bfcc272afc9c756ed1a67b157945c9bb5bc4/src/components/users/user-signin/ko/userSigninViewModelBinder.ts

## Expected behavior

The ReturnUrl should be the current URL in the developer portal.

## Is your portal managed or self-hosted?

Managed

## Release tag or commit SHA (if using self-hosted version)
commit `34f3bfcc272afc9c756ed1a67b157945c9bb5bc4`

## API Management service name

fzag-apim-int

## Environment

 - Operating system: Windows 10
 - Browser: Edge
 - Version: 79.0.309.71

## Additional context

![image](https://user-images.githubusercontent.com/8446397/74031271-53ab5d80-49b1-11ea-953d-059b6bd2c048.png)

**Comment by nelkenson** [comment]
We're having the same issue. Is there a workaround?

**Comment by mikebudzynski** [comment]
The only workaround at this moment is to self-host the developer portal.

**Comment by mikebudzynski** [comment]
Update: this will be fixed in the managed portal in the upcoming release (1-3 weeks).

**Comment by MFeldinho** [comment]
Hello! Any updates on this? :)

**Comment by samuel-weber** [comment]
This issue is very critical to us. Since 3 weeks are over now, we would really like to know, when this issue will be fixed and released (managed version).

**Comment by mikebudzynski** [comment]
The issue is already fixed, but not yet released. Our deployments are currently on hold due to the global emergency state, hopefully they will resume within a month.

**Comment by samuel-weber** [comment]
> The issue is already fixed, but not yet released. Our deployments are currently on hold due to the global emergency state, hopefully they will resume within a month.

Hi @mikebudzynski: Is the issue also fixed in the "Self-hosted code"? If I look into the file mentioned by @RetoRechsteiner . The mentioned code seems not to be changed - [signinViewModelBinder.ts](https://github.com/Azure/api-management-developer-portal/blob/master/src/components/users/signin/ko/signinViewModelBinder.ts)

**Comment by samuel-weber** [comment]
By the way it seems that signin-sso does not work at all (anymore). 
If I browse the site using signin-sso. 

The user is **not** "really" logged in -Just the menu thinks the user is logged in:

![image](https://user-images.githubusercontent.com/7022900/79311158-4bb1f180-7efd-11ea-990e-fc886e7440e4.png)

When I then click the on the Profile Menu - The user will be redirected to the /login page :-(


I generated the url using this function: https://docs.microsoft.com/en-us/rest/api/apimanagement/2019-12-01/User/GenerateSsoUrl

This gave this url (witch is the legacy portal by the way):
https://<my-service-name>.portal.azure-api.net/signin-sso?token=xyz...

I then changed the url by replacing ".portal." to ".developer."
https://<myservice>.developer.azure-api.net/signin-sso?token=xyz...


--> Everything works with the legacy portal (also redirect) this is really frustrating!

**Comment by ygrik** [comment]
Hello @samuel-weber 
This is the known issue: the token for **".developer."** is encoded, but for the **".portal."** must be without encoding. 
Just decode token for **".developer."** and use it for **".portal."**
This issue will be fixed with the next release for the managed version.

**Comment by guideveloper** [comment]
> By the way it seems that signin-sso does not work at all (anymore).
> If I browse the site using signin-sso.
> 
> The user is **not** "really" logged in -Just the menu thinks the user is logged in (see Reports-menu):
> 
> ![image](https://user-images.githubusercontent.com/7022900/79311158-4bb1f180-7efd-11ea-990e-fc886e7440e4.png)
> 
> When I then click the on the Profile Menu - The user will be redirected to the /login page :-(
> 
> tl;dr;
> I generated the url using this function: https://docs.microsoft.com/en-us/rest/api/apimanagement/2019-12-01/User/GenerateSsoUrl
> 
> This gave this url (witch is the legacy portal by the way):
> https://(my-service-name).portal.azure-api.net/signin-sso?token=xyz...
> 
> I then changed the url by replacing ".portal." to ".developer." to point to the new developer portal
> https://(my-service-name).developer.azure-api.net/signin-sso?token=xyz...
> 
> --> Everything works with the legacy portal (also redirect) this is really frustrating!

@samuel-weber Did you get this resolved please, Im trying to let the users login via the API management azure backend but the signin-sso end point isn't setup on my self hosted portal and not sure what to do with the token that gets generated?

**Comment by mikebudzynski** [comment]
We're actively working on a solution. cc: @ygrik

**Comment by mikebudzynski** [comment]
The fix for this issue is included in the release we're rolling out now. ETA for full deployment is ~2 weeks.

**Comment by mikebudzynski** [comment]
The fix for this issue is included in the release we're rolling out now. ETA for full deployment is ~2 weeks.
